File: VB-Examples/07_DocumentOperation/SetAuthorForRevisions.vb
Imports System.IO
Imports System.Text
Imports Spire.Doc
Imports Spire.Doc.Documents
Imports Spire.Doc.Fields

Namespace SetAuthorForRevisions
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new Document object
			Dim document As New Document()

			' Load the document from the specified file path
			document.LoadFromFile("..\..\..\..\..\..\..\Data\ExtractText.docx")

			'Get the first section
			Dim section As Section=document.Sections(0)

			' Start track revisions
			document.StartTrackRevisions("test")

			' Set author for deleted revision
			Dim para As Paragraph =document.LastParagraph
			para.Text = ""
			For i As Integer = 0 To para.ChildObjects.Count - 1
				Dim textRange As TextRange = TryCast(para.ChildObjects(i), TextRange)
				If textRange.IsDeleteRevision Then
					textRange.DeleteRevision.Author = "user1"
				End If
			Next i

			' Set author for inserted revision
			Dim paragraph As Paragraph = section.AddParagraph()
			Dim range As TextRange = paragraph.AppendText("Added text")
			range.InsertRevision.Author = "user2"

			' Stop track revisions
			document.StopTrackRevisions()

			' Save the file
			document.SaveToFile("SetAuthorForRevisions_out.docx", FileFormat.Docx)

			' Dispose of the Document object 
			document.Dispose()

		
			WordDocViewer("SetAuthorForRevisions_out.docx")

		End Sub
		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS-Examples/07_DocumentOperation/SetAuthorForRevisions.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;
using Spire.Doc.Fields;

namespace SetAuthorForRevisions
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new Document object
            Document document = new Document();

            // Load the document from the specified file path
            document.LoadFromFile(@"..\..\..\..\..\..\..\Data\ExtractText.docx");

            //Get the first section
            Section section=document.Sections[0];

            // Start track revisions
            document.StartTrackRevisions("test");

            // Set author for deleted revision
            Paragraph para =document.LastParagraph;
            para.Text = "";
            for (int i = 0; i < para.ChildObjects.Count; i++)
            {
                TextRange textRange = para.ChildObjects[i] as TextRange;
                if (textRange.IsDeleteRevision)
                {
                    textRange.DeleteRevision.Author = "user1";
                }
            }

            // Set author for inserted revision
            Paragraph paragraph = section.AddParagraph();
            TextRange range = paragraph.AppendText("Added text");
            range.InsertRevision.Author = "user2";

            // Stop track revisions
            document.StopTrackRevisions();  

            // Save the file
            document.SaveToFile("SetAuthorForRevisions_out.docx", FileFormat.Docx);

            // Dispose of the Document object 
            document.Dispose();


            WordDocViewer("SetAuthorForRevisions_out.docx");

            this.Close();
        }
        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
